File: enchantgui/gui/src/main/java/cc/mewcraft/enchantment/gui/gui/ForwardItem.java
package cc.mewcraft.enchantment.gui.gui;

import cc.mewcraft.enchantment.gui.config.EnchantGuiSettings;
import cc.mewcraft.enchantment.gui.util.AdventureUtils;
import com.google.inject.Inject;
import org.bukkit.Material;
import xyz.xenondevs.invui.gui.PagedGui;
import xyz.xenondevs.invui.item.ItemProvider;
import xyz.xenondevs.invui.item.builder.ItemBuilder;

public class ForwardItem extends AnimatedPageItem {

    @Inject
    public ForwardItem(EnchantGuiSettings settings) {
        super(settings, true);
    }

    @Override public ItemProvider getItemProvider(final PagedGui<?> gui) {
        ItemBuilder builder = new ItemBuilder(Material.SPECTRAL_ARROW);
        builder.setDisplayName(AdventureUtils.translatable("menu.enchantment.next_page"));
        return builder;
    }
}

File: enchantgui/gui/src/main/kotlin/cc/mewcraft/enchantment/gui/gui/ForwardItem.kt
package cc.mewcraft.enchantment.gui.gui

import cc.mewcraft.enchantment.gui.config.EnchantGuiSettings
import cc.mewcraft.enchantment.gui.util.translatable
import cc.mewcraft.enchantment.gui.util.wrapper
import com.google.inject.Inject
import org.bukkit.Material
import xyz.xenondevs.invui.gui.PagedGui
import xyz.xenondevs.invui.item.ItemProvider
import xyz.xenondevs.invui.item.builder.ItemBuilder

class ForwardItem
@Inject constructor(
    settings: EnchantGuiSettings,
) : AnimatedPageItem(settings, true) {
    override fun getItemProvider(gui: PagedGui<*>): ItemProvider =
        ItemBuilder(Material.SPECTRAL_ARROW).apply {
            displayName = "menu.enchantment.next_page".translatable().wrapper()
        }
}
